### PR TITLE
Add support for ERC-721 and ERC-1155

### PR DESCRIPTION
**Pull request by pscott**

### CODE REVIEW

File: src_plugins/erc1155/erc1155_provide_parameters.c [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+}
+
+static void handle_batch_transfer(ethPluginProvideParameter_t *msg, erc1155_context_t *context) {
+    switch (context->next_param) {
+        case FROM:
+            context->next_param = TO;
+            break;
+        case TO:
+            copy_address(context->address, msg->parameter, sizeof(context->address));
+            context->next_param = TOKEN_ID;
```
Comment (greenknot) [inline src_plugins/erc1155/erc1155_provide_parameters.c:34]: It should be `TOKEN_IDS_OFFSET`, right?

File: src/eth_plugin_handler.c [hunk 2]
```diff
@@ -54,61 +72,83 @@ void eth_plugin_prepare_query_contract_UI(ethQueryContractUI_t *queryContractUI,
...
+    }
+    if (memcmp(init->selector,
+               dataContext.tokenContext.methodSelector,
+               sizeof(dataContext.tokenContext.methodSelector)) != 0) {
+        PRINTF("Got selector: %.*H\n", SELECTOR_SIZE, init->selector);
+        PRINTF("Expected selector: %.*H\n", SELECTOR_SIZE, dataContext.tokenContext.methodSelector);
+        os_sched_exit(0);
+    }
+    PRINTF("Plugin will be used\n");
+    // TODO: Add check for chainid.
```
Comment (greenknot) [inline src/eth_plugin_handler.c:95]: Please add this check.

File: src_features/provideNFTInformation/cmd_provideNFTInfo.c [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+
+    PRINTF("Length: %d\n", collectionNameLength);
+    PRINTF("CollectionName: %s\n", nft->collectionName);
+    offset += collectionNameLength;
+
+    memcpy(nft->contractAddress, workBuffer + offset, ADDRESS_LENGTH);
+    PRINTF("Address: %.*H\n", ADDRESS_LENGTH, workBuffer + offset);
+    offset += ADDRESS_LENGTH;
+
+    // TODO: store chainID and assert that tx is using the same chainid.
```
Comment (greenknot) [inline src_features/provideNFTInformation/cmd_provideNFTInfo.c:132]: Please fix this TODO.

**Comment by apaillier-ledger** [comment]
Superseded by #218 & #229.

**Pull request closed without merging**